builder/virtualbox/ovf/builder.go

```diff
@@ -48,6 +51,20 @@ func (b *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packe
 	state.Put("hook", hook)
 	state.Put("ui", ui)
 
+	// Convert source path to downloadable url
+	sourcePath, err := common.DownloadableURL(b.config.SourcePath)
+	if err != nil {
+		return nil, fmt.Errorf("Error preparing source_path: %s", err)
+	}
+
+	// Unless already specified, override the targetPath whenever dealing with OVAs
```

Comment:
if you leave TargetPath empty in StepDownload, it'll actually set up the cache correctly, using the Url as the key. Check out https://github.com/mitchellh/packer/blob/master/packer/cache.go and see it being used here https://github.com/mitchellh/packer/blob/master/common/step_download.go#L69-L83